### DISCUSSION TITLE
Create custom viewer with sound notification

### DISCUSSION THREAD

**Discussion by manosrh** [cpvalente/ontime#933] [General]
I'm asking for some help on creating a **custom viewer** (using Ontime-viewer-template-v2 (https://github.com/cpvalente/Ontime-viewer-template-v2 maybe).
But I need to enhance a new feature:
--> I need the browser to **play a sound when the countdown reaches one minute before finish**.

The application is amazing! I understand that this feature is not in the team's priorities, but I think it will be useful for many people!

Is it possible to make it work with custom HTML, CSS and JS?

**Comment by cpvalente** [comment]
Hi @manosrh, thank you for reaching out!

The logic to achieve this should be straightforward, in the websocket endpoint ([which is connected in the template](https://github.com/cpvalente/Ontime-viewer-template-v2/blob/main/app.js#L69)) you see the current property of the timer. That is equivalent to the value of the running timer (in milliseconds)

You would only need to check if the value received triggers your flag and start the sound using a audio API.
If you have some code to share I am happy to fill in the gaps 

Quick reminder that this will change with v3 and I will make a new template once v3 is out

**Reply by manosrh** [reply]
Hello @cpvalente and thank you for your reply!
I already use v3 beta! I still have no code to share, I was planning to start tomorrow.
Is it better to wait for the new template v3?

**Reply by cpvalente** [reply]
Good to hear you are using v3, if you have any feedback it would be great to hear.

I dont think you need to wait, the v2 template will still get you running, just the shape of the data has changed.
You can see the documentation for the data here
https://docs.getontime.no/api/data/runtime-data/

specifically what you are looking for will be in send in an object like

```ts
{
  type: "ontime-timer",
  payload: {
    current <---- 
    ...
  }
}
```
